### PR TITLE
 Add Set-PowerBIWorkspace cmdlet

### PR DESCRIPTION
**Pull request by nandemoii**
- The Set-PowerBIWorkspace cmdlet works for the Organization scope; the Individual scope is not supported yet
- Add NotImplementedException as a terminating error
- Add test utilities classes to reduce test code duplication
- Fix local build script error

### CODE REVIEW

File: src/NuGet.config [hunk 1]
```diff
@@ -1,7 +1,7 @@
 <?xml version="1.0" encoding="utf-8"?>
 <configuration>
   <packageSources>
-    <add key="ChazShare" value="\\cmb-pbi2\Public\Nugets" />
+    <add key="private-nuget-package" value="C:\git\nugetpackages" />
```
Comment (speshoria) [inline src/NuGet.config]: I don't think we need to check-in this change.

File: src/Modules/Workspaces/Commands.Workspaces/help/Set-PowerBIWorkspace.md [hunk 2]
```diff
@@ -0,0 +1,108 @@
+---
+external help file: Microsoft.PowerBI.Commands.Workspaces.dll-Help.xml
+Module Name: MicrosoftPowerBIMgmt.Workspaces
```
Comment (speshoria) [inline src/Modules/Workspaces/Commands.Workspaces/help/Set-PowerBIWorkspace.md]: Update this file with description and examples

File: src/Modules/Workspaces/Commands.Workspaces.Test/SetPowerBIWorkspaceTests.cs [hunk 3]
```diff
@@ -0,0 +1,131 @@
...
+                    { "Id", new Guid() },
+                    { "Name", "Updated Workspace Name" },
+                    { "Description", "Updated Workspace Description" },
+                };
+                ps.AddCommand(Cmdlet).AddParameters(parameters);
+
+                try
+                {
+                    ps.Invoke();
+                }
```
Comment (CodeCyclone) [inline src/Modules/Workspaces/Commands.Workspaces.Test/SetPowerBIWorkspaceTests.cs]: Add Assert.Fail() below

File: src/Modules/Workspaces/Commands.Workspaces.Test/SetPowerBIWorkspaceTests.cs [hunk 4]
```diff
@@ -0,0 +1,131 @@
...
+
+                var result = ps.Invoke();
+
+                Assert.Fail("Should not have reached this point");
+            }
+        }
+
+        private static Group GetWorkspace(System.Management.Automation.PowerShell ps, string id = null)
+        {
+            ps.AddCommand(new CmdletInfo($"{GetPowerBIWorkspace.CmdletVerb}-{GetPowerBIWorkspace.CmdletName}", typeof(GetPowerBIWorkspace))).AddParameter("Scope", "Organization");
```
Comment (CodeCyclone) [inline src/Modules/Workspaces/Commands.Workspaces.Test/SetPowerBIWorkspaceTests.cs]: Could refactor this to a common spot too

File: src/Modules/Workspaces/Commands.Workspaces/SetPowerBIWorkspace.cs [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+            if (Uri.TryCreate(this.Profile.Environment.GlobalServiceEndpoint, UriKind.Absolute, out Uri baseUri))
+            {
+                client = new PowerBIClient(baseUri, new TokenCredentials(token.AccessToken));
+            }
+            else
+            {
+                client = new PowerBIClient(new TokenCredentials(token.AccessToken));
+            }
+            
+            if (this.Scope.Equals(PowerBIUserScope.Individual))
```
Comment (CodeCyclone) [inline src/Modules/Workspaces/Commands.Workspaces/SetPowerBIWorkspace.cs]: Can move this up to the top

File: src/Modules/Workspaces/Commands.Workspaces/SetPowerBIWorkspace.cs [hunk 6]
```diff
@@ -0,0 +1,67 @@
...
+
+        [Parameter(Mandatory = true)]
+        public Guid Id { get; set; }
+
+        [Parameter(Mandatory = false)]
+        public string Name { get; set; }
+
+        [Parameter(Mandatory = false)]
+        public string Description { get; set; }
+
```
Comment (CodeCyclone) [inline src/Modules/Workspaces/Commands.Workspaces/SetPowerBIWorkspace.cs]: We should support passing in a Group as a parameter too. We should define ParameterSets

File: src/NuGet.config (hunk 1 above)
Comment (CodeCyclone) [reply src/NuGet.config]: I'm OK with it as its not pointing to my share anymore and always local

File: src/Modules/Workspaces/Commands.Workspaces/SetPowerBIWorkspace.cs [hunk 7]
```diff
@@ -14,30 +14,42 @@
...
-        [Parameter(Mandatory = false)]
+        [Parameter(Mandatory = false, ParameterSetName = PropertiesParameterSetName)]
         public string Name { get; set; }
 
-        [Parameter(Mandatory = false)]
+        [Parameter(Mandatory = false, ParameterSetName = PropertiesParameterSetName)]
         public string Description { get; set; }
 
+        [Parameter(Mandatory = true, ParameterSetName = WorkspaceParameterSetName)]
+        public Group Workspace { get; set; }
```
Comment (CodeCyclone) [inline src/Modules/Workspaces/Commands.Workspaces/SetPowerBIWorkspace.cs]: Add an [Alias("Group")] just for backward compat

**Pull request merged**